### PR TITLE
Add DroidKaigi 2021 Article item in FeedScreen

### LINKED ISSUE #622: Add DroidKaigi 2021 Article item in FeedScreen
## Kind (Required)

- Improvement

 ## Overview (Required)

- We want to make it easy for users to navigate to the DroidKaigi2021 timetable from the FeedScreen.
- Right now I'm showing the first Feed article, but I'd like to show DroidKaigi2021 until November.
image

 ## Links	

- https://www.figma.com/file/IFlrbfmBSdYvUz7VmSzfLV/DroidKaigi_2021_official_app?node-id=2313%3A772

### PR DESCRIPTION
**Pull request by Aqua-ix**
## Issue
- close #622 

## Overview (Required)
- Added a link to the DroidKaigi2021 timetable as the first article on FeedScreen.

## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
:eyes:

**Comment by takahirom** [comment]
I think this item is very important for DroidKaigi.
Therefore, I think it can be removed from the logic of the favorites button and filter. 👍

**Comment by takahirom** [comment]
@nontan 
I'm going to remove the favorite button from the article that leads to DroidKaigi2021 that I've added to the article list. This is because this article is special and important for DroidKaigi and because the logic for keeping favorites is also complex. 🙇 
記事一覧に表示する今回追加したDroidKaigi2021への誘導記事からお気に入りボタンを消しちゃおうと思っています。この記事はDroidKaigiにとって特別で重要なのとお気に入りを保持するロジックも複雑になるためです。 🙇

File: model/src/commonMain/kotlin/io/github/droidkaigi/feeder/Media.kt [hunk 1]
```diff
@@ -4,13 +4,15 @@ sealed class Media(val text: String) {
     object YouTube : Media("YouTube")
     object DroidKaigiFM : Media("DroidKaigi.FM")
     object Medium : Media("Medium")
+    object Conference: Media("Conference")
```
Comment (takahirom) [inline model/src/commonMain/kotlin/io/github/droidkaigi/feeder/Media.kt:7]: Plesae apply ` ./gradlew spotlessKotlinApply` 🙏

File: uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/DroidKaigi2021ArticleItem.kt [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+import io.github.droidkaigi.feeder.core.theme.typography
+import kotlinx.datetime.Instant
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.toLocalDateTime
+
+@Composable
+fun DroidKaigi2021ArticleItem(
+    favorited: Boolean,
+    onClick: () -> Unit,
+    onFavoriteChange: (FeedItem) -> Unit,
```
Comment (takahirom) [inline uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/DroidKaigi2021ArticleItem.kt:39]: We can remove favorite button, so we can remove `onFavoriteChange: (FeedItem) -> Unit,` and `favorited: Boolean,` 👍

File: uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedScreen.kt [hunk 3]
```diff
@@ -317,32 +322,29 @@ private fun FeedList(
...
                     if (isFilterState) {
                         FilterItemCountRow(feedContents.size.toString())
                     }
-                    FirstFeedItem(
-                        feedItem = feedItem,
-                        favorited = favorited,
-                        onClick = onClickFeed,
-                        showMediaLabel = isHome,
-                        onFavoriteChange = onFavoriteChange,
+                    DroidKaigi2021ArticleItem(
```
Comment (takahirom) [inline uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedScreen.kt:329]: The point I wrote in this issue is that I want to remove DroidKaigi2021ArticleItem in November. That's why I thought we will need to refactor. 💭 But we can create issue for that 👍 

Like this

```kotlin
object DroidKaigi2021 {
  fun isArticleEnd() = instant.isAfter(...)
}

val isDroidKaigiEnd = remember { mutableStateOf { DroidKaigi2021.isArticleEnd() } }

if (!isDroidKaigiEnd) {
    DroidKaigi2021ArticleItem(...)
} else {
...
}
```

https://github.com/DroidKaigi/conference-app-2021/issues/622
 `I'd like to show DroidKaigi2021 until November`

**Review by takahirom — COMMENTED** [review]
Almost LGTM! Please check the comments 🙏

**Comment by nontan** [comment]
@takahirom I understand! Thank you for everything.

File: model/src/commonMain/kotlin/io/github/droidkaigi/feeder/Media.kt (hunk 1 above)
Comment (Aqua-ix) [reply model/src/commonMain/kotlin/io/github/droidkaigi/feeder/Media.kt]: fixed: 90e7981

File: uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/DroidKaigi2021ArticleItem.kt (hunk 2 above)
Comment (Aqua-ix) [reply uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/DroidKaigi2021ArticleItem.kt]: Removed favorite button: 7b98ad7

File: uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedScreen.kt (hunk 3 above)
Comment (Aqua-ix) [reply uicomponent-compose/feed/src/main/java/io/github/droidkaigi/feeder/feed/FeedScreen.kt]: Refactored at ca0bc3d
The logic is a bit complicated ...

**Comment by Aqua-ix** [comment]
@takahirom 
Thank you for your review.
Fixed what you pointed out.

**Review by takahirom — APPROVED** [review]
LGTM 👍 Thank you for thinking with me!  This function is very important for us! 🙏

**Comment by takahirom** [comment]
I think I can do it later. But if you're interested, could you take a look? 🙏 
https://github.com/DroidKaigi/conference-app-2021/issues/718

**Comment by Aqua-ix** [comment]
@takahirom 
Thx for reviewing! ありがとうございました！
I'd like to try #718.

**Pull request merged**